src/protocol/produce.rs

```diff
@@ -24,19 +27,22 @@ pub struct ProduceRequest<'a, 'b> {
     pub timeout: i32,
     pub topic_partitions: Vec<TopicPartitionProduceRequest<'b>>,
     pub compression: Compression,
+    pub api_version: Rc<ApiVersion>,
```

Comment:
Two questions:

1. Why are these `ApiVersion`s being added to these request structs? If I am not mistaken, the API version goes into the request *header*, not the request itself, so I'm not sure it makes sense to have the API version stored inside every request object.
2. Why are they all `Rc`? This adds significant overhead on every access. `ApiVersion` is just a simple struct, so even copying it is quite cheap.